src/components/NetworkStatus/NetworkStatus.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C)  Online-Go.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import * as React from "react";
+import { socket } from "sockets";
+
+//interface NetworkStatusProps {}
+
+export function NetworkStatus(): JSX.Element {
+    const [hidden, setHidden] = React.useState(true);
+
+    React.useEffect(() => {
+        socket.on("latency", () => {
+            setHidden(true);
+        });
+        socket.on("timeout", () => {
+            setHidden(false);
+        });
+        socket.on("disconnect", () => {
+            setHidden(false);
+        });
```

Comment:
Probably doesn't have consequence in this case since this component is only mounted once, but it's good practice to unsubscribe from events in the [cleanup function](https://legacy.reactjs.org/docs/hooks-effect.html#effects-with-cleanup) of `useEffect`:

```
useEffect(() => {
    const show = () => setHidden(false);
    const hide = () => setHidden(true);

    socket.on("latency", hide);
    socket.on("timeout", show);
    socket.on("disconnect", show);

    return () => {
        socket.off("latency", hide);
        socket.off("timeout", show);
        socket.off("disconnect", show);
    }
}, [])
```